x-pack/plugins/infra/public/components/asset_details/tabs/profiling/flamegraph.tsx

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useMemo } from 'react';
+import { i18n } from '@kbn/i18n';
+import { EmbeddableFlamegraph } from '@kbn/observability-shared-plugin/public';
+import { EuiEmptyPrompt } from '@elastic/eui';
+import { useAssetDetailsRenderPropsContext } from '../../hooks/use_asset_details_render_props';
+import { useDatePickerContext } from '../../hooks/use_date_picker';
+import { useProfilingFlamegraphData } from '../../hooks/use_profiling_flamegraph_data';
+import { useTabSwitcherContext } from '../../hooks/use_tab_switcher';
+import { ContentTabIds } from '../../types';
+
+export function Flamegraph() {
+  const { asset } = useAssetDetailsRenderPropsContext();
+  const { activeTabId } = useTabSwitcherContext();
+  const { getDateRangeInTimestamp } = useDatePickerContext();
+  const { from, to } = getDateRangeInTimestamp();
+
+  const params = useMemo(
+    () => ({
+      hostname: asset.name,
+      from,
+      to,
+    }),
+    [asset.name, from, to]
+  );
+  const { error, loading, response } = useProfilingFlamegraphData({
+    isActive: activeTabId === ContentTabIds.PROFILING,
+    params,
+  });
+
+  if (error !== null) {
+    return <ErrorPrompt />;
+  }
+
+  return <EmbeddableFlamegraph data={response ?? undefined} isLoading={loading} height="60vh" />;
```

Comment:
Is it really necessary to add this `response ?? undefined`

Reply:
Unfortunately yes, `useHTTPRequest` hook has `response: T | null` and `EmbeddableFlamegraph` has `data?: T`. This needs to be unified by changing either the hook or embeddable, but I did not want to start this refactoring as part of this change as it might blow out of control 😅